### ISSUE TITLE
[BUG] Specifying schema for TPH entities causes malformed migrations

### ISSUE THREAD

**Issue by jelleverheyen** [efcore/EFCore.NamingConventions#34]
**EF Core Version:**  3.1.5
**Npgsql.EntityFrameworkCore.PostgreSQL version:** 3.1.4
**EFCore.NamingConventions version**: 1.1.0
**dotnet-ef tool version:** 3.1.6

## Behavior
Using the `NamingConventions.UseSnakeCaseNamingConvention()`;

We have a base class `Permission` with 2 subclasses `OperationPermission` and `DataPermission` defined as follows:
```
modelBuilder
    .Entity<Permission>()
    .ToTable("permissions", "auth")
    .HasDiscriminator(x => x.Type)
    .HasValue<OperationPermission>(AlphaClaimTypes.Permission)
    .HasValue<DataPermission>(AlphaClaimTypes.PermissionOnData);
```

When creating a migration, the model in the Designer file will be generated as follows:
```
modelBuilder.Entity("Common.Authorization.Persistence.DataPermission", b =>
{
    b.HasBaseType("Common.Authorization.Persistence.Permission");
    ,"auth");

    b.HasDiscriminator().HasValue("PermissionOnData");
});

modelBuilder.Entity("Common.Authorization.Persistence.OperationPermission", b =>
{
    b.HasBaseType("Common.Authorization.Persistence.Permission");
    ,"auth");

    b.HasDiscriminator().HasValue("Permission");
});
```

The lines below `b.HasBaseType();` are `,"auth"`, which cannot be compiled.
This behavior is also replicated in the `ContextModelSnapshot` and consistent across TPH hierarchies.

Removing the usage of `NamingConventions.UseSnakeCaseNamingConvention()` solves this problem.

These types are as follows:
```cs
public abstract class Permission
    {
        public int Id { get; set; }
        public string Type { get; set; }
        public string Value { get; set; }
        public string Data { get; set; }
        public string DataHash { get; set; } 

        public ICollection<PermissionToPermissionGroup> PermissionGroups { get; set; }
    }

    public class OperationPermission : Permission
    {
        protected OperationPermission() { }

        public OperationPermission(string operation)
        {
            Type = AlphaClaimTypes.Permission;
            Value = operation;
        }
    }

    public class DataPermission : Permission
    {
        protected DataPermission() { }

        public DataPermission(string name, SortedDictionary<string, string> properties)
        {
            Type = AlphaClaimTypes.PermissionOnData;
            Value = name;
            Data = DataPermissionKey.Pack(properties);
            DataHash = DataPermissionKey.Hash(Data);
        }
    }
```

## Expected Behavior
A schema or table name should not be assigned to a discriminated entity, only the model of the base class has to define this.

**Comment by roji** [comment]
Thanks for reporting this.

What is the exact type of Type? This doesn't seem to be the .NET ClaimTypes, is it?

Also, can you please give the [5.0.0-preview5](https://www.nuget.org/packages/EFCore.NamingConventions/5.0.0-preview5) version a test and report if it works any better?

**Comment by jelleverheyen** [comment]
@roji I just updated the issue with the types.
No it's not .NET ClaimTypes, it's something we implemented on our end:
```cs
public static class AlphaClaimTypes
    {
        public const string Permission = nameof(Permission);
        public const string PermissionOnData = nameof(PermissionOnData);
    }
```

Later today I will set up a way to reproduce and one with 5.0.0-preview5 as you mentioned

**Comment by roji** [comment]
@jelleverheyen any progress on this? I just released 5.0.0-preview8, it would be great if you could try on that.

**Comment by roji** [comment]
Confirmed that this is gone in 5.0.0-rc1.
